### ISSUE TITLE
does not work with mongoose 5.x.

### ISSUE THREAD

**Issue by bonesoul** [buunguyen/mongoose-deep-populate#53]
```
  - promise.js:65 check
    []/[bluebird]/js/release/promise.js:65:15

  - promise.js:72 new Promise
    []/[bluebird]/js/release/promise.js:72:9

  - plugin.js:98 createMongoosePromise
    []/[mongoose-deep-populate]/lib/plugin.js:98:17

  - plugin.js:67 model.Query.Query.exec
    []/[mongoose-deep-populate]/lib/plugin.js:67:14

```

**Comment by yamifr07** [comment]
```
TypeError: expecting a function but got [object Undefined]
    at check (/Users/yamilafraiman/Desktop/projects/inamu/webserver/node_modules/bluebird/js/release/promise.js:65:15)
    at new Promise (/Users/yamilafraiman/Desktop/projects/inamu/webserver/node_modules/bluebird/js/release/promise.js:72:9)
    at createMongoosePromise (/Users/yamilafraiman/Desktop/projects/inamu/webserver/node_modules/mongoose-deep-populate/lib/plugin.js:98:17)
    at model.Query.Query.exec (/Users/yamilafraiman/Desktop/projects/inamu/webserver/node_modules/mongoose-deep-populate/lib/plugin.js:67:14)
```

**Comment by buunguyen** [comment]
Released 3.1
